tests/components/sensor/test_foobot.py

```diff
@@ -0,0 +1,75 @@
+"""The tests for the Foobot sensor platform."""
+
+import re
+import sys
+import unittest
+
+import homeassistant.components.sensor as sensor
+from homeassistant.setup import setup_component
+from homeassistant.const import (TEMP_CELSIUS)
+from tests.common import (
+    get_test_home_assistant, load_fixture, assert_setup_component)
+
+from ...test_util.aiohttp import mock_aiohttp_client
+
+VALID_CONFIG = {
+    'platform': 'foobot',
+    'token': 'adfdsfasd',
+    'username': 'example@example.com',
+}
+
+
+# pylint: disable=invalid-name
+class TestFoobotSetup(unittest.TestCase):
+    """Test the setup portion of the Foobot platform."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+        self.config = VALID_CONFIG
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @unittest.skipIf(sys.version_info < (3, 5),
+                     "Test not working on Python 3.4")
```

Comment:
Why is this test not working on Python 3.4? This component should all currently supported versions of Home Assistant, and that includes Python 3.4...

(Adding a few more test cases for a platform might not be too bad either...)

Reply:
Unfortunately, I haven't been able to make the tests work under 3.4. AsyncIO testing under 3.4 seems to require a bit of specific work (event loop related) and considering support for 3.4 in HASS will be deprecated in a few weeks, I didn't feel like putting the work towards that.

I'm fine if it means delaying the merging of this platform until post 0.64.

Reply:
We have dropped Python 3.4. Can you remove this skip and can you resolve the merge conflict?

Reply:
I agree with @balloob :)

Reply:
I've installed it and it works... but with the limitation of 200 APIs calls :(

Reply:
Any idea to solve tjis problem? Anyhone is working about this subject to include this interesting gadget as a new component in Home Assistant?

Reply:
Unfortunately, the 200 API calls/day is enforced by foobot and there isn't much I can do about this without special permission. 

On the other hand, 200 API calls/day is enough to get data every 10 minutes, averaged over this  period and it is usable. The netatmo sensors works at this same frequency for example.

Reply:
Sure, 200 API calls/day would be enough. My problem is that using it for about 4 hours I spent more than 125 calls. Why?

Reply:
I'll check, this version has been working fine for me for a while, I'm sure it did restrict the amount of requests correctly with ~0.64 but I haven't updated my running instance to 0.65 yet.

Reply:
For me it seems 6 API calls / 10 minutes

Reply:
This is strange. Would you happen to have the RESTful sensor component still hitting the API?

I've updated this PR shortly with the changes requested.Could I ask you to check the logs?

First, you'll need to activate debug logs for this component:

```
logger:
  logs:
    homeassistant.components.sensor.foobot: debug
```

And relaunch hass.

What you should see in `homeassistant.log` is:

`DEBUG (MainThread) [homeassistant.components.sensor.foobot] The data response is: [{'time': 1521039027, 'pm': 134.00002, 'tmp': 21.597, 'hum': 46.16466, 'co2': 1121.6666, 'voc': 310.0, 'allpollu': 128.69048}]` At launch and once every ten minutes. It'll display the number of requests since launch, excluding the initial request to get the devices as `DEBUG (MainThread) [homeassistant.components.sensor.foobot] Number of requests: 1`

It should be followed by 5 `DEBUG (MainThread) [homeassistant.components.sensor.foobot] Skipping update`. Every request is logged so it should be evident if it's hitting foobot servers too frequently.